lib: getopt: Fix compiler warning.

koffes opened the pull request:
current_dash may be used uninitialized

Fixes #57520

koffes committed:
lib: getopt: Fix compiler warning.

current_dash may be used uninitialized

Signed-off-by: Kristoffer Rist Skøien <kristoffer.skoien@nordicsemi.no>

cfriedt reviewed (changes requested):
@koffes - thanks for providing the fix. I'm actually unable to reproduce the issue using any of our existing testsuite configurations, but upon inspection, it's clear why this variable should be initialized.

If possible, could you or @Frodevan provide reproducibility details for the `tests/posix/getopt/` testsuite in #57520? E.g. was a static analysis tool used? Can the output of the static analyzer be added to the description?

cfriedt reviewed (changes requested):
@koffes - one minor request just to ensure that we *do* have coverage for this in CI, and thank you for your patience.

Can you please add a second commit to this PR with the change below?
```diff
diff --git a/samples/bluetooth/iso_broadcast/sample.yaml b/samples/bluetooth/iso_broadcast/sample.yaml
index fdd0b34436..be80a0ad65 100644
--- a/samples/bluetooth/iso_broadcast/sample.yaml
+++ b/samples/bluetooth/iso_broadcast/sample.yaml
@@ -14,3 +14,12 @@ tests:
       - nrf52dk_nrf52832
     extra_args: OVERLAY_CONFIG=overlay-bt_ll_sw_split.conf
     tags: bluetooth
+  # to ensure that the issue reported in #57520 has coverage in CI
+  sample.bluetooth.iso_broadcast.getopt:
+    platform_allow: nrf5340_audio_dk_nrf5340_cpuapp
+    build_only: True
+    tags: bluetooth
+    extra_configs:
+      - CONFIG_COMPILER_WARNINGS_AS_ERRORS=y
+      - CONFIG_GETOPT=y
+      - CONFIG_GETOPT_LONG=y
```

cc @jakub-uC - would be great to identify why this was not caught by the `tests/posix/getopt` testsuite, since `CONFIG_COMPILER_WARNINGS_AS_ERRORS=y` is set by `twister`. Any ideas why that might be the case @nashif? It's a bit puzzling to me. Additional details [here](https://github.com/zephyrproject-rtos/zephyr/issues/57520#issuecomment-1536440041) and [here](https://github.com/zephyrproject-rtos/zephyr/issues/57520#issuecomment-1537519807)

Thalley commented on lib/posix/getopt/getopt_long.c:
```diff
@@ -172,7 +172,7 @@ parse_long_options(struct getopt_state *state, char * const *nargv,
 	char *current_argv, *has_equal;
 #ifdef GNU_COMPATIBLE
-	char *current_dash;
+	char *current_dash = "";
```
Suggest to also remove 
```c
		default:
			current_dash = "";
break;
```
in the switch below, as that code is now redundant

jakub-uC replied:
The root cause is explained in: https://github.com/zephyrproject-rtos/zephyr/issues/57520#issuecomment-1538130307

koffes replied:
I have removed `current_dash = ""` but the default switch case is required by the code standard.

Commit:
getopt: extend ztests scope

Adding a test to detect problems in the cooperation of the getopt
and logger modules. It would detect issue: #57520

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>

jakub-uC commented:
The root cause is explained in: https://github.com/zephyrproject-rtos/zephyr/issues/57520#issuecomment-1538130307

cfriedt commented:
@koffes - again, sorry for the hoops, but it would be preferable if you could replace your second commit to `samples` with the change to `testcase.yaml` made in #56752.

With that, we have both the fix and the added test coverage in the same PR which is ideal for backports.

Thanks!

koffes commented:
@cfriedt : I don't follow. In https://github.com/zephyrproject-rtos/zephyr/pull/56752 there is no mention of testcase.yaml.

jakub-uC commented:
@koffes I think @cfriedt pointed out the wrong PR (the link is broken).

Please refer to this: #57652

koffes commented:
There we go @jakub-uC and @cfriedt :)
